Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -43,15 +43,51 @@ class MessageContentView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
     var filePreviewView: _ChatMessageFileAttachmentListView<ExtraData>?
     var giphyView: _ChatMessageGiphyView<ExtraData>?
     var actionsView: _ChatMessageInteractiveAttachmentView<ExtraData>?
-    var reactionsView: _ReactionsCompactView<ExtraData>?
     var errorIndicatorView: _ChatMessageErrorIndicator<ExtraData>?
 
+    var reactionsView: _ReactionsCompactView<ExtraData>?
+    var reactionsBubbleView: ReactionsBubbleView<ExtraData>?
+
     var threadReplyCountButton: UIButton?
     var threadAvatarView: ChatAvatarView?
     var threadArrowView: _SimpleChatMessageThreadArrowView<ExtraData>?
     
     lazy var mainContainer: ContainerView = ContainerView(axis: .horizontal)
         .withoutAutoresizingMaskConstraints
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        maskContainerByReactionsBubble()
+    }
+
+    func maskContainerByReactionsBubble() {
+        guard let reactionsBubble = reactionsBubbleView else {
+            mainContainer.mask = nil
+            return
+        }
+
+        let bubbleOriginInMainContainer = reactionsBubble.superview!
+            .convert(reactionsBubble.frame.origin, to: mainContainer)
+
+        let moveToMainContainer = CGAffineTransform(
+            translationX: bubbleOriginInMainContainer.x,
+            y: bubbleOriginInMainContainer.y
+        )
+
+        let conainerMaskingPath = reactionsBubble.maskingPath
+        conainerMaskingPath.apply(moveToMainContainer)
+
+        let maskImage = UIGraphicsImageRenderer(size: mainContainer.bounds.size)
+            .image {
+                let layer = CAShapeLayer()
+                layer.path = conainerMaskingPath.cgPath
+                layer.render(in: $0.cgContext)
+            }
+            .asAlphaMask()
```

Comment:
This part is computationally heavy and will be replaced later when we figure out how to build a mask with `evenOdd` + `UIBezierPath` with the intersections (reactions bubble's tail has 2 overlapping circles and we have a hole in mask where circles intersect)